### ISSUE TITLE
Add settings/preferences for multiple mailboxes & custom domains

### ISSUE THREAD

**Issue by JeGr** [simple-login/browser-extension#36]
As I'm trying to setup SimpleLogin for multiple mailboxes and custom domains, what really stood out was the missing features of the browser extension vs. e.g. the smartphone apps. In case of importance that are:

1. No option to even select the mailbox that alias has to go to! That's the most important one with more than one mailbox configured and I was really curious why that is missing while it's the topmost option in the app versions?
2. There's no option to turn on/off the PGP toggle (that is missing from the apps, too). If someone has multiple mailboxes configured in the backend and only some offer PGP while others don't (e.g. your example on the website with protonmail & gmail), it's a messy thing to manually go over every alias created to check for PGP on/off toggles. 
3. A convenience thing: No option to toggle the default/preferred domain in the extension. In my usecase the person operating e.g. the tech-mailbox will need to mainly create aliases for the tech.domain.xyz custom domain in the account while another operating the dev-mailbox will mainly use dev.domain.xyz as its custom domain. So it would be a great help not only being able to switch between them but selecting a preferred one for new aliases per browser/API key.
4. There's no option to setup the "alias response" setting of the newly created alias so when responding you have a name/tag/teamname configured

I'd really like to see the extension more on par with the app one in term of functionality as having to check on every laptop/desktop about the correct setting of e.g. PGP setting, the setup of "Alias Name" or more important the correct mailbox would be a really messy process ;)

Happy to help if there's anything I can supply.

Best Regards,
Jens

**Comment by ngxson** [comment]
Hi Jens,

Thanks for your feedback. I've been the active developer for SimpleLogin's browser extension for 1 month now. I understand that the extension lacks for many features, compared to web or mobile version. We are still trying to reconstruct and add many more features, such as having a button next to email input fields to quickly generate a new alias.

However, due to the large number of features, adding all at once can make the extension becomes unstable (a.k.a buggy). That's why we are working on a "Beta" channel of the extension. For now, we are still waiting for approval from Chrome and Firefox.

After the release of this "Beta" channel, I'll try to implement features that you mentioned. @nguyenkims will surely send you an email for more details so you can get the beta version as soon as possible.

Thanks again for your very detailed information. Have a good day!

**Comment by JeGr** [comment]
Hi!

First, thanks for the quick follow up and an apology heads-up as I didn't check first how "young" the extension still is. Somehow I just assumed (falsely as I regret), that it has been around for quite some time what kept me wondering about those things I pointed out. So sorry for jumping the gun here ;)

> That's why we are working on a "Beta" channel of the extension. For now, we are still waiting for approval from Chrome and Firefox.

Definetly something I'd like to use and help you with reports as the actual coding is not my strong suite ;) But testing and usability are something I can help with :)

If you need more in-depth information about any of the points I can further relate what/why but so far if sth like a beta build is in the works I'll happily wait for it and be on board first thing it arrives.

So thanks for letting me know and sorry for assuming more "age" to the extension as it is. For that quick build time of a month, that's really nicely done!

A quick "bug" I'd like to add: if you're logged in the Dashboard of Simplelogin with the browser that you installed the extension in, it is "logged in" (assuming via user/pass credentials) but still creates an API token like "Chrome (32bit)" that is actually never used (checked by deleting it - still was logged in, could create aliases etc.).
While trying to switch the extension to another API key I manually created, I logged off in the extension (click on the wheel, Logout) and was then automatically logged out in the browser tab itself, too. After logging in, there again was the Chrome 32bit token and my manually created one as well. I then copied my manual token, logged out of the extension again, logged in via API method and used my manual token and logged in in the browser tab again. Then deleted the Chrome 32bit token and tested the extension again: now the API token I created showed actual usage instead of the one created "automagically" that was named Chrome 32bit and showed "never used" that irritated me ;) Perhaps something in the login handling/usage when logged into the Dashboard in another tab or something to do with my browser not being Chrome but Brave (based on Chrome) or "Edgeium" (extension was tested in Brave only).

Can try to play that again with Edgium if that would help narrow it down.

Best regards,
\jens

**Comment by ngxson** [comment]
Yes, I have just noticed the same problem too. I believe that there's something to do with the backend.

@nguyenkims For short: API tokens created from cookie always shows "never used" on the usage section. Can you take a look? Thanks!

**Comment by nguyenkims** [comment]
@JeGr @ngxson thanks for the info! The bug has been fixed in https://github.com/simple-login/app/pull/246. 

@JeGr there's still a lot of missing features on the chrome/firefox extension compared to the web and mobile apps and our current goal is to get it on par with the others :). 

1), 2), 4) are already in our todo list.

I just added https://trello.com/c/045Lz9ug/107-able-to-select-preferred-domain-when-creating-new-a-new-alias-per-browser-api-key for 3), feel free to vote on it to make it happen sooner :)

As soon as the beta passes the review, I'll let you know!

**Comment by JeGr** [comment]
@nguyenkims Thanks for the quick fix. So the extension shouldn't create an unnecessary API key anymore or it should use it now? ;) I'd test that later, so would be nice in advance which result I should see to report back :) 

As for the trello list: do I have to register for that separately or is there some bridge in place to the simplelogin account?

**Comment by nguyenkims** [comment]
@JeGr the extension will automatically create an API key if user is already logged in. It might feel redundant but as the cookie-based authentication can be fragile within an extension, we choose to use API Key to be on the safe side :). In the future, we might remove this mechanism.

In order to vote on Trello, you'd need to sign up a Trello account I think. You can create an alias to sign up a Trello account of course 🙃

**Comment by JeGr** [comment]
Did already create the account by using one of my simplelogin custom domains ;)

Yeah I guessed that the extension created the API key itself but in my case it didn't use it but used the cookie/login so it showed that API key as never used before. That's what got me curious. I'll try again with another browser/laptop later and see what happens now ;)

**Comment by JeGr** [comment]
To follow up with the extension: is there a beta version already available as a developer sideload or sth alike? Would love to get testing if there's already work done :)

**Comment by nguyenkims** [comment]
@JeGr we submitted the beta version to Chrome and Firefox store last week and are waiting for their review :). Will let you know as soon as we have the green light :).

**Comment by nguyenkims** [comment]
@JeGr the beta version is finally approuved on Chrome that you can install at https://chrome.google.com/webstore/detail/simplelogin-beta-your-ant/flalafddooibiagdacankdmiaojekdbk?hl=en&authuser=0

This version brings in a lot of features, hope you'll like it :)

**Comment by JeGr** [comment]
Seen and installed it. Already have a few comments. Should I leave them here or open a new thread/topic for the beta extension to better separate things?

**Comment by nguyenkims** [comment]
@JeGr Thanks for taking the time to test it! It would be great if you can create a separate Github issue for each (category of) feedback. But you can also create a new Github issue with all feedbacks if separating them doesn't make much sense :). 

There's also the "Report an issue" button if an issue is specific to a website.
![image](https://user-images.githubusercontent.com/3167780/90263172-93b33780-de4f-11ea-854d-c20f8c4c8910.png)

**Comment by aosavi** [comment]
Is there any news on the OP's first case (changing the mailbox in the (Firefox) extension)? Is there a beta version that implements it? Thanks!

**Comment by nguyenkims** [comment]
@aosavi currently we haven't implemented the mailbox change in the extension yet.

**Comment by TheConen** [comment]
The Proton Pass extension has this functionality. Since Simplelogin is part of Proton, can we expect this to arrive in the Simplelogin extension soon?

**Comment by Dubz** [comment]
Expanding on 3:

Allow us more flexibility by allowing us to change the domain entirely, and choose whether the random alias is words or UUID based.

Right now, the only way to change this is to open the dashboard and change the defaults. I think having this setting local instead of cloud based is the best approach. Last domain used is default selection (but changeable, of course) plus 3 buttons (words, UUID, custom) just like it is on the website. The custom input value can default to the site name/domain like it does now.
